File: Selenium/Activities/activity14.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
import java.util.List;

public class activity14 {
    public static void main(String args[]) {
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/tables");
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        System.out.println("Title of the page is: " + driver.getTitle());
        List<WebElement> rows = driver.findElements(By.xpath("//table[@id='sortableTable']/tbody/tr"));
        System.out.println("Number of rows: "+rows.size());
        List<WebElement> cols = driver.findElements(By.xpath("//table[@id='sortableTable']/tbody/tr[1]/td"));
        System.out.println("Number of columns: "+cols.size());
        WebElement secondCellValue = driver.findElement(By.xpath("//table[@id='sortableTable']/tbody/tr[2]/td[2]"));
        System.out.println("Second row second Cell value is: "+secondCellValue.getText());
        driver.findElement(By.xpath("//table[@id='sortableTable']/thead/tr/th")).click();
        WebElement secondCellValueAgain = driver.findElement(By.xpath("//table[@id='sortableTable']/tbody/tr[2]/td[2]"));
        System.out.println("Second row second Cell value is: "+secondCellValueAgain.getText());
        List<WebElement> footercells = driver.findElements(By.xpath("//table[@id='sortableTable']/tfoot/tr/th"));
        for(WebElement cell:footercells) {
            System.out.println(cell.getText());
        }
        driver.close();
    }
}

File: Selenium/Activities/activity14.py
from selenium.webdriver.common.by import By
from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
driver = webdriver.Firefox(service=FirefoxService(GeckoDriverManager().install()))
driver.get("https://v1.training-support.net/selenium/tables")
print("Page title is: ", driver.title)
columns = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/thead/tr/th")
print("Number of columns: ", len(columns))
rows = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/tbody/tr")
print("Number of rows: ", len(rows))
second_row_second_cell_before_sort = driver.find_element(By.XPATH, "//table[@id='sortableTable']/tbody/tr[2]/td[2]")
print("Cell value at second row and second column: ", second_row_second_cell_before_sort.text)
driver.find_element(By.XPATH, "//table[@id='sortableTable']/thead/tr/th[1]").click()
second_row_second_cell_after_sort = driver.find_element(By.XPATH, "//table[@id='sortableTable']/tbody/tr[2]/td[2]")
print("Cell value at second row and second column: ", second_row_second_cell_after_sort.text)
driver.close()
